shell/platform/darwin/ios/framework/Headers/FlutterPlatformViews.h

```diff
@@ -0,0 +1,63 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_FLUTTERPLATFORMVIEWS_H_
+#define FLUTTER_FLUTTERPLATFORMVIEWS_H_
+
+#import <CoreMedia/CoreMedia.h>
+#import <Foundation/Foundation.h>
+#import <UIKit/UIKit.h>
+
+#include "FlutterCodecs.h"
+#include "FlutterMacros.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ An instance of a `UIView` created by a `FlutterPlatformViewFactory`.
+
+ This protocol is used to associate a view with a dispose callback.
+ */
+FLUTTER_EXPORT
+@protocol FlutterPlatformView <NSObject>
```

Comment:
```suggest

This is a confusing name.  We already have an internal `PlatformViewIOS`.  maybe
```suggestion
@protocol FlutterWrappedUIView <NSObject>
```

Reply:
Yeah that's a good point, and it goes beyond this specific protocol but applies to the entire concept of "platform (iOS/Android) views". We are dealing with 2 kinds of platform views, the platform views into which we are rendering Flutter pixels, and the platform views we embed. I guess the other obvious alternative was to call the later embedded views, but I'm not sure it would have been much better as the term embedded also has 2 meanings 😄  (the code that embeds Flutter to a platform, and the code that embeds a platform view to Flutter).

I went with platform views early on which had a single meaning in framework land, I'd vote for renaming it everywhere only if we come up with a really good name as the rename will involve some amount of friction (this term is currently used in the framework, in the Android platform views implementation, in other places in the engine, and I've even seen a third party medium post about Flutter platform views).

Another alternative we might want to consider here is instead of having this protocol, to add a `disposeViewWithId` method to `FlutterPlatformViewFactory`, though we would still have a **FlutterPlatformView**Factory, and it will be inconsistent with the way we currently do things on Android (I guess a breaking change to the Android code is kind of still possible but will be an annoyance to at least some users).

Reply:
Part of my concern here is this is now leaking out into framework-public code.  Developers will be more likely to interact with it, and renaming it will get that much harder (whereas someone who is dealing with a PlatformView today is working more internally to the engine).

Reply:
After offline discussion, I think I agree that this is more of an internal problem that we can resolve with some better documentation for the Engine's class hierarchy (and maybe a rename of some not-as-public members).